redfish-core/lib/log_services.hpp

```diff
@@ -6719,27 +6719,144 @@ inline void getRedfishUriByDbusObjPath(
+ * @param[in]   services    List of services to exporting the D-bus object path
+ * @param[in] entryJsonIdx  The json entry index to add isolated hardware
+ *                          details in the appropriate entry json object.
+ * @param[in]   guardType   The guardtype value as a string. To be used in
+ *                          the redfish message property
+ *
+ * @return void
+ */
+inline void updateHwIsolationMessage(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& path, const dbus::utility::MapperServiceMap& services,
+    const size_t entryJsonIdx, const std::string& guardType)
+{
+    // Ensure we only got one service back
+    if (services.size() != 1)
+    {
+        BMCWEB_LOG_ERROR("Invalid Service Size {}", services.size());
+        for (const auto& service : services)
+        {
+            BMCWEB_LOG_ERROR("Invalid Service Name: {}", service.first);
+        }
+        if (asyncResp)
+        {
+            messages::internalError(asyncResp->res);
+        }
+        return;
+    }
+    sdbusplus::asio::getProperty<std::string>(
+        *crow::connections::systemBus, services[0].first, path,
+        "xyz.openbmc_project.Inventory.Item", "PrettyName",
+        [asyncResp, path, entryJsonIdx,
+         guardType](const boost::system::error_code& ec,
+                    const std::string& prettyName) {
+        loadHwIsolationMessage(ec, prettyName, asyncResp, path, entryJsonIdx,
+                               guardType);
+    });
+}
+
+/**
+ * @brief API used to get "PrettyName" by using the given dbus object path
+ *        and fill into "Message" property of LogEntry schema.
+ *
+ * @param[in] asyncResp .   The redfish response to return.
+ * @param[in] dbusObjPath   The DBus object path which represents redfishUri.
+ * @param[in] entryJsonIdx  The json entry index to add isolated hardware
+ *                            details in the appropriate entry json object.
+ * @param[in]   guardType   The guardtype value as a string. To be used in
+ *                          the redfish message property
+ * @return void
  */
 
 inline void getPrettyNameByDbusObjPath(
     const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
     const sdbusplus::message::object_path& dbusObjPath,
-    const size_t entryJsonIdx)
-{
-    crow::connections::systemBus->async_method_call(
-        [asyncResp, dbusObjPath,
-         entryJsonIdx](const boost::system::error_code& ec,
-                       const dbus::utility::MapperGetObject& objType) mutable {
+    const size_t entryJsonIdx, const std::string& guardType)
+{
+    constexpr std::array<std::string_view, 1> interface = {
+        "xyz.openbmc_project.Inventory.Item"};
+    dbus::utility::getDbusObject(
+        dbusObjPath.str, interface,
+        [asyncResp, dbusObjPath, entryJsonIdx,
+         guardType](const boost::system::error_code& ec,
+                    const dbus::utility::MapperGetObject& objType) mutable {
```

Comment:
why this change?  is it improvement over existing code or was it not working

Reply:
This is the upstream style. An improvement, we tend to incorporate whenever we touch the code